xray/plotting.py

```diff
@@ -237,9 +237,9 @@ def wrapper(darray, ax=None, xincrease=None, yincrease=None,
                              'Passed DataArray has {} dimensions'
                              .format(plotfunc.__name__, len(darray.dims)))
 
-        x = darray[xlab]
-        y = darray[ylab]
-        z = darray
+        x = darray[xlab].values
```

Comment:
`pcolormesh` doesn't work with non-ndarrays -- it tries to call the `.ravel()` method, which currently doesn't exist on DataArray objects.